src/variables.jl

```diff
@@ -790,3 +790,15 @@ function all_variables(model::Model)
         model, MOI.ListOfVariableIndices())::Vector{MOI.VariableIndex}
     return VariableRef[VariableRef(model, idx) for idx in all_indices]
 end
+
+"""
+    dual(vref::VariableRef)
+
+Returns a proper error saying that the user should refer to UpperBoundRef,
+LowerBoundRef or FixRef to get the reduced dual from a constraint.
+"""
```

Comment:
This function doesn't need a doctstring since it just implements a helpful error fallback.